client/dashboard/sites/settings-repositories/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { HostingFeatures } from '@automattic/api-core';
+import { siteBySlugQuery, codeDeploymentsQuery } from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import { __experimentalText as Text, Button } from '@wordpress/components';
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo } from 'react';
+import { siteRoute } from '../../app/router/sites';
+import { Callout } from '../../components/callout';
+import { CalloutOverlay } from '../../components/callout-overlay';
+import { DataViewsCard } from '../../components/dataviews-card';
+import PageLayout from '../../components/page-layout';
+import UpsellCTAButton from '../../components/upsell-cta-button';
+import { hasHostingFeature } from '../../utils/site-features';
+import illustrationUrl from '../deployments/deployments-callout-illustration.svg';
+import ghIconUrl from '../deployments/gh-icon.svg';
+import SettingsPageHeader from '../settings-page-header';
+import { useRepositoryFields } from './dataviews/fields';
+import { DEFAULT_VIEW, DEFAULT_LAYOUTS } from './dataviews/views';
+import type { CodeDeploymentData } from '@automattic/api-core';
+import type { View } from '@wordpress/dataviews';
+
+export function SiteRepositoriesCallout( {
+	siteSlug,
+	titleAs = 'h1',
+}: {
+	siteSlug: string;
+	titleAs?: React.ElementType | keyof JSX.IntrinsicElements;
+} ) {
+	return (
+		<Callout
+			icon={ <img src={ ghIconUrl } alt={ __( 'GitHub logo' ) } /> }
+			title={ __( 'Deploy from GitHub' ) }
+			titleAs={ titleAs }
+			image={ illustrationUrl }
+			description={
+				<>
+					<Text as="p" variant="muted">
+						{ __(
+							'Connect your GitHub repo directly to your WordPress.com site—with seamless integration, straightforward version control, and automated workflows.'
+						) }
+					</Text>
+					<Text as="p" variant="muted">
+						{ __( 'Available on the WordPress.com Business and Commerce plans.' ) }
+					</Text>
+				</>
+			}
+			actions={
+				<UpsellCTAButton
+					text={ __( 'Upgrade plan' ) }
+					tracksId="deployments"
+					variant="primary"
+					href={ `/checkout/${ siteSlug }/business` }
+				/>
+			}
+		/>
+	);
+}
+
```

Comment:
This callout is identical to the one on the Deployments page [here](https://github.com/Automattic/wp-calypso/blob/6377980381883e4f8f610a020bfe97107f85baac/client/dashboard/sites/deployments/index.tsx#L18-L53). I plan to extract this to a separate component and import it back in both places after https://github.com/Automattic/wp-calypso/pull/105651 is merged.